src/server/transaction.cc

```diff
@@ -493,9 +493,7 @@ bool Transaction::RunInShard(EngineShard* shard, bool txq_ooo) {
   unsigned idx = SidToId(shard->shard_id());
   auto& sd = shard_data_[idx];
 
-  bool prev_armed = sd.is_armed.load(memory_order_relaxed);
-  DCHECK(prev_armed);
-  sd.is_armed.store(false, memory_order_relaxed);
+  CHECK(sd.is_armed.exchange(false, memory_order_relaxed));
```

Comment:
Please also add `CHECK_GT(run_count_.load(acquire), 0)` - lets see if we catch it earlier.

Reply:
maybe relaxed? If we change acquisition semantics we might change the code